### PR TITLE
Semantic search notebooks

### PR DESCRIPTION
**Pull request by bovlb**
This contains the two halves of the Wikipedia demo, ingest and search.
Web crawl version to follow shortly.

### CODE REVIEW

File: examples/semantic_search/cohere_wikipedia_ingest.ipynb [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    "See our [Configuration](https://docs.aperturedata.io/Setup/client/configuration) help page for more options."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! adb config create rag --from-json --active "
```
Comment (vishakha041) [inline examples/semantic_search/cohere_wikipedia_ingest.ipynb:146]:
```suggestion
    "! adb config create rag_adb_server_config  --from-json --active "
```

Comment (vishakha041) [reply examples/semantic_search/cohere_wikipedia_ingest.ipynb:146]: I am thinking we should do more descriptive config names for the configs OR how does it play with the one in JSOn already?

File: examples/semantic_search/cohere_wikipedia_search.ipynb [hunk 2]
```diff
@@ -0,0 +1,516 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "55683ea8",
+   "metadata": {},
+   "source": [
+    "# Building a RAG chain from Wikipedia\n",
+    "\n",
+    "This notebook shows how to use ApertureDB as part of a Retrieval-Augmented Generation Langchain pipeline.  This means that we're going to use ApertureDB as a vector-based search engine to find documents that match the query and then use a large-language model to generate an answer based on those documents. \n",
```
Comment (vishakha041) [inline examples/semantic_search/cohere_wikipedia_search.ipynb:10]: Link to APerture / langchain page?

File: examples/semantic_search/cohere_wikipedia_search.ipynb [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+    "        run_query(user_query)\n",
+    "    \n",
+    "def run_query(user_query):\n",
+    "    display(Markdown(f\"### User Query\\n{user_query}\"))\n",
+    "        \n",
+    "    nonrag_answer = plain_chain.invoke(user_query)\n",
+    "    display(Markdown(f\"### Non-RAG Answer\\n{nonrag_answer}\"))\n",
+    "\n",
+    "    rag_answer = rag_chain_with_source.invoke(user_query)\n",
+    "    display(Markdown(\"\\n\".join([\n",
```
Comment (vishakha041) [inline examples/semantic_search/cohere_wikipedia_search.ipynb:458]: How about putting the answer as well?

File: examples/semantic_search/cohere_wikipedia_search.ipynb [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+   "source": [
+    "# Building a RAG chain from Wikipedia\n",
+    "\n",
+    "This notebook shows how to use ApertureDB as part of a Retrieval-Augmented Generation Langchain pipeline.  This means that we're going to use ApertureDB as a vector-based search engine to find documents that match the query and then use a large-language model to generate an answer based on those documents. \n",
+    "\n",
+    "We already have a corpus of >600k paragraphs from the Simple English Wikipedia with associated embeddings provided by Cohere.\n",
+    "(If not, see [Ingesting Wikipedia into ApertureDB](./cohere_wikipedia_ingest.ipynb)).\n",
+    "We'll use that to answer natural-language questions.\n",
+    "\n",
+    "![RAG workflow](images/RAG_Demo.png)\n",
```
Comment (vishakha041) [inline examples/semantic_search/cohere_wikipedia_search.ipynb:16]: Ok a few suggestions on the image to make it tie better to what we need:
1. Either put logo or explicitly say ApertureDB vector store and document store
2. swap the blue and green - Blue is our color. green is all outside stuff. 
3. YOu can also name the models you have used  Embedding model e.g. Cohere xyz , LLM e.g. openai gpt4o 

Some optional suggestions: maybe lose the 3d ness and straighten / align arrows

**Review by vishakha041 — CHANGES_REQUESTED** [review]
Overall I like the content of these notebooks. Just some suggestions for finishing touches.

Bigger concern: we now have duplicates in demo and here?
IS there a reason you had to add them to python repo instead of leaving them in the demo repo?  Is it for colab to work?
I am not sure we should load up our Python SDK repo with large app examples. For now, how about we leave this in demo (so you can just put the neat nice notebooks there) and we land them in docs as an article. Then we figure out how to open source demo repo in the next couple weeks so we can link those notebooks for opening in colab at that point?

File: examples/semantic_search/cohere_wikipedia_ingest.ipynb (hunk 1 above)
Comment (bovlb) [reply examples/semantic_search/cohere_wikipedia_ingest.ipynb:146]: We will be able to drop the name entirely once https://github.com/aperture-data/aperturedb-python/pull/510 is released.

File: examples/semantic_search/cohere_wikipedia_search.ipynb (hunk 3 above)
Comment (bovlb) [reply examples/semantic_search/cohere_wikipedia_search.ipynb:458]: I'm not sure what you mean.  Do you want me to include human-generated answers for these questions?

File: examples/semantic_search/cohere_wikipedia_search.ipynb (hunk 4 above)
Comment (bovlb) [reply examples/semantic_search/cohere_wikipedia_search.ipynb:16]: Vishakha will take on improvements to this image.

**Comment by bovlb** [comment]
Moved to new repo

**Pull request closed without merging**